Sources/RoutingKit/Routing/TrieRouter.swift

```diff
@@ -70,11 +70,17 @@ public final class TrieRouter<Output>: CustomStringConvertible {
             // check the constants first
             for constant in currentNode.constants {
                 let match: Bool
-                if ci {
-                    match = constant.value.utf8.elementsEqual(caseInsensitive: path.utf8)
+                
+                // Short circuit early if items are different lengths
+                if constant.value.count != path.count {
+                    match = false
+                } else if ci {
```

Comment:
We should rename `ci` here to `isCaseInsensitive` to better match the other code.